impl/src/test/java/org/jboss/forge/test/roaster/model/JavaInterfaceMethodTest.java

```diff
@@ -0,0 +1,154 @@
+      method = javaInterface.getMethods().get(javaInterface.getMethods().size() - 1);
+   }
+
+   @Test
+   public void testGetMethodByString()
+   {
+      javaInterface.addMethod("public void random() { }");
+      javaInterface.addMethod("public void random(String randomString) { }");
+
+      MethodSource<JavaInterfaceSource> randomMethod = javaInterface.getMethod("random");
+
+      List<ParameterSource<JavaInterfaceSource>> randomMethodParameters = randomMethod.getParameters();
+      assertEquals(0, randomMethodParameters.size());
+      assertFalse(javaInterface.hasMethodSignature(method.getName()));
+
+      MethodSource<JavaInterfaceSource> randomMethodString = javaInterface.getMethod("random", "String");
+      List<ParameterSource<JavaInterfaceSource>> randomMethodStringParameters = randomMethodString.getParameters();
+      assertEquals(1, randomMethodStringParameters.size());
+      assertEquals("String", randomMethodStringParameters.get(0).getType().getName());
+      assertFalse(javaInterface.hasMethodSignature(method.getName()));
+   }
+
+   @Test
+   public void testGetMethodByClass() throws Exception
+   {
+      javaInterface.addMethod("public void random() { }");
+      javaInterface.addMethod("public void random(String randomString) { }");
+
+      MethodSource<JavaInterfaceSource> randomMethod = javaInterface.getMethod("random");
+      List<ParameterSource<JavaInterfaceSource>> randomMethodParameters = randomMethod.getParameters();
+      assertEquals(0, randomMethodParameters.size());
+      assertFalse(javaInterface.hasMethodSignature(method.getName()));
+
+      MethodSource<JavaInterfaceSource> randomMethodString = javaInterface.getMethod("random", String.class);
+      List<ParameterSource<JavaInterfaceSource>> randomMethodStringParameters = randomMethodString.getParameters();
+      assertEquals(1, randomMethodStringParameters.size());
+      assertEquals("String", randomMethodStringParameters.get(0).getType().getName());
+      assertFalse(javaInterface.hasMethodSignature(method.getName()));
+   }
+
+   @Test
+   public void testSetAbstract()
+   {
+      assertTrue(method.isAbstract());
+      method.setAbstract(true);
+      Assert.assertEquals(((MethodDeclaration) method.getInternal()).getModifiers(), Modifier.ABSTRACT);
+      exception.expect(IllegalStateException.class);
+      method.setAbstract(false);
+   }
+
+   @Test
+   public void testSetConstructor()
+   {
+      assertFalse(method.isConstructor());
+      exception.expect(UnsupportedOperationException.class);
+      method.setConstructor(true);
+   }
+
+   @Test
+   public void testSetPublic()
+   {
+      assertTrue(method.isPublic());
+      method.setPublic();
+      Assert.assertEquals(((MethodDeclaration) method.getInternal()).getModifiers(), Modifier.PUBLIC);
+
+   }
+
+   @Test
+   public void testSetPrivate()
+   {
+      assertFalse(method.isPrivate());
+      exception.expect(UnsupportedOperationException.class);
```

Comment:
Any reason to use this Rule? @Test(expected=UnsupportedOperationException) seems to do the same